src/main/java/gregtech/api/Situation.java

```diff
@@ -0,0 +1,41 @@
+package gregtech.api;
+
+import gregtech.api.util.GTControlledRegistry;
+
+public class Situation {
+
+    public static GTControlledRegistry<String, Situation> SITUATION_REGISTRY = new GTControlledRegistry<>(Short.MAX_VALUE);
+    private final String situationName;
+    public Enum situationType;
+    public String situationLocaleName;
```

Comment:
For situation name we should recommend using snake_case and provided name directly use in lang files. This will need some prefixing but it should save us need for another param